content/en/methods/apps.md

```diff
@@ -25,20 +22,53 @@ POST /api/v1/apps HTTP/1.1
 
 Create a new application to obtain OAuth2 credentials.
 
-**Returns:** [Application]({{< relref "entities/Application" >}})\
+{{< hint style="danger" >}}
+In Mastodon prior to 4.3, OAuth Applications could be "vacuumed" and removed from the database under certain conditions, meaning your Application's `client_id` and `client_secret` would not be recognised by the Mastodon server.\
+This automated removal of applications was removed in Mastodon 4.3\
+\
+A workaround for Mastodon versions older than 4.3 was to register your application, and then immediately request a [Client Credential]({{< relref "client/Token#flow" >}}) token, which would permanently ensure your application always had an active access token and would not be removed.
+{{< /hint >}}
+
+{{< hint style="info" >}}
+At present, Mastodon only supports provisioning confidential clients, i.e., you will always receive a `client_secret` and `client_secret_expires_at` attributes in the [CredentialApplication]({{< relref "entities/Application#CredentialApplication" >}}) entity.\
+\
+For more information see: [OAuth 2 client types]({{< relref "spec/oauth#client-types" >}})
+{{< /hint >}}
+
+**Returns:** [CredentialApplication]({{< relref "entities/Application#CredentialApplication" >}})\
 **OAuth:** Public\
 **Version history:**\
 0.0.0 - added\
-2.7.2 - now returns `vapid_key`
+2.7.2 - now returns `vapid_key`\
+4.3.0 - deprecated `vapid_key`, please see [api/v2/instance]({{< relref "methods/Instance#v2">}})\
+4.3.0 - added support for multiple `redirect_uris` in Form data parameters\
+4.3.0 - added `redirect_uris` response property\
+4.3.0 - deprecated `redirect_uri` response property, since this can be a non-URI if multiple `redirect_uris` are registered, use `redirect_uris` instead\
+4.3.0 - changed entity type from [Application]({{< relref "entities/Application">}}) to [CredentialApplication]({{< relref "entities/Application#CredentialApplication">}})
+
+#### Request {#create-request-example}
+
+Example request:
+
+```
+POST /api/v1/apps HTTP/1.1
+Content-Type: application/json
+
+{
+  "client_name": "Test Application",
+  "redirect_uris": ["https://app.example/callback", "https://app.example/register"],
+  "scopes": "read write push",
+  "website": "https://app.example"
+}
+```
 
-#### Request
 ##### Form data parameters
 
 client_name
 : {{<required>}} String. A name for your application
 
 redirect_uris
-: {{<required>}} String. Where the user should be redirected after authorization. To display the authorization code to the user instead of redirecting to a web page, use `urn:ietf:wg:oauth:2.0:oob` in this parameter.
+: {{<required>}} String or Array of Strings. Where the user should be redirected after authorization. To display the authorization code to the user instead of redirecting to a web page, use `urn:ietf:wg:oauth:2.0:oob` in this parameter.
```

Comment:
"String or Array of Strings" probably needs to be clarified. It's a String when passed in as a form data parameter, and an Array of String when using JSON, right? I think it should say "String" here, and then add a note at the end to the effect of `If using the JSON format for the API, multiple URIs should be passed in as an Array of String.`

Reply:
I think you can also do:

```
curl -X POST \
	-F 'client_name=Test Application' \
	-F 'redirect_uris[]=urn:ietf:wg:oauth:2.0:oob' \
	-F 'redirect_uris[]=https://app.example/callback' \
	-F 'scopes=read write push' \
	-F 'website=https://myapp.example' \
	https://mastodon.example/api/v1/apps
```

Reply:
Without the `[]` on `redirect_uris` in the curl example, it'll use the last passed line, in this case `https://app.example/callback` (though, I'm not sure from which spec the []= syntax originates from)